### ISSUE TITLE
Support non-JSP templates

### ISSUE THREAD

**Issue by kngenie** [iipc/openwayback#281]
Wayback implements its own web application framework, and it is heavily dependent on JSP as view templates: It doesn't offer good support beans for other template frameworks. JSP paths are specified directly in bean properties, and scattered around the code. _Renderer_ concept is inconsistent among different types of view.

Possible redesign idea:
* Converge on single interface for view rendering. Spring's `View` appears to be a good candidate
* Redesign `UIResults` to support other template frameworks. Pass model objects through View interface, instead of UIResults.extract* methods.
* Add more support methods and beans, fix method signatures for easier access from templates (ex. `UIResults.getCount` instead of `UIResults.count`.)

**Comment by kris-sigur** [comment]
I agree, it is probably worth switching this out wholesale for a more standard solution.
